src/cortex.js

```diff
@@ -44,9 +44,29 @@ module.exports = (function() {
       }
     }
 
+    __remove(path) {
+      for(var i = 0, ii = this.__diffs.length; i < ii; i++) {
```

Comment:
I'm not sure I follow the logic here. It's a little inefficient with O(n^2). It seems the simplest solution is to just compare the initial and final state. If they're the same the rollback/cancel the update

Reply:
You mean loop _once_ through all queued diffs in `this._diffs` comparing them to `this.__value`?
If so, you mean, here, inside the `__remove()` event handler, right?

Reply:
Instead of using diffs, it’s probably better to let the update runs its course and as a final check perform a deep comparison between final and initial values (they’re immutable so already separate objects). If they’re the same then don’t run the callbacks.